### PR TITLE
Storybook upgrade + components refactor: part 1

### LINKED ISSUE #3907: Dependencies(components): Storybook replace deprecated code
Issues to solve with Storybook used in **components** package:

- [ ] update to newest versions `7` https://storybook.js.org/blog/storybook-7-0/
- [ ] replace deprecated [@storybook/addon-knobs](https://storybook.js.org/addons/@storybook/addon-knobs/) package by [@storybook/addon-controls](https://storybook.js.org/addons/@storybook/addon-controls/)
- [ ] get rid of deprecated  `storiesOf` from  `@storybook/react` library
- [x] fix controlled components used in form elements
- [ ] Solve problem with losing focus in Input and Textarea stories 

Get inspired by other [Storybook projects](https://storybook.js.org/docs/react/get-started/examples)

... To be continued

### PR DESCRIPTION
**Pull request by komret**
### Storybook upgrade to v7
The upgrade was done using the `npx storybook@latest upgrade` script provided in the [Storybook migration guide](https://storybook.js.org/docs/react/migration-guide). I added `storyStoreV7: false` to `main.js` config file to keep support for old `storiesOf` syntax. I refactored components using the newer, CSF 2 syntax, to the latest, CSF 3 syntax, which is somewhat simpler. I partially used a codemod provided in the migration guide to do it.
Closes #3907

TODO:
- refactor the remaining stories to CSF 3 syntax
- remove the deprecated `knobs` addon in favour of `controls` addon

#### Refactor components
I mainly aimed at renaming the index files, but couldn't help but do some more changes. I added two stories and made some small improvements and refactoring in the `PinButton` component.

TODO:
- rename the remaining index files
- revise package folder structure

### CODE REVIEW

**Comment by tomasklim** [comment]
Related to #3907

File: packages/components/src/components/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import styled from 'styled-components';
```
Comment (dahaca) [inline packages/components/src/components/Dropdown/Dropdown.stories.tsx:2]: I would also remove [those 2 rules](https://github.com/trezor/trezor-suite/blob/c9c7d7db09e51a8cc50891b6fc383d1c3aacc6f9/packages/components/tsconfig.json#L8) as a part of Part 2? @tomasklim Removing the `components` package from the blacklist would be the next step, many more rules come in actions if that's done.

**Review by dahaca — APPROVED** [review]

File: packages/components/src/components/Dropdown/Dropdown.stories.tsx (hunk 1 above)
Comment (tomasklim) [reply packages/components/src/components/Dropdown/Dropdown.stories.tsx:2]: I agree

File: packages/components/package.json [hunk 2]
```diff
@@ -40,22 +40,25 @@
...
-        "@storybook/addon-controls": "^6.5.16",
-        "@storybook/addon-knobs": "^6.4.0",
-        "@storybook/addon-links": "^6.5.16",
-        "@storybook/addon-viewport": "^6.5.16",
-        "@storybook/addons": "^6.5.16",
-        "@storybook/builder-webpack5": "^6.5.16",
-        "@storybook/client-api": "^6.5.16",
+        "@babel/preset-env": "^7.22.9",
+        "@babel/preset-react": "^7.22.5",
+        "@babel/preset-typescript": "^7.22.5",
```
Comment (tomasklim) [inline packages/components/package.json]: Those are in the root `package.json`. I guess we do not need it here or do we?

File: packages/components/.babelrc.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "sourceType": "unambiguous",
+    "presets": [
+        [
+            "@babel/preset-env",
+            {
+                "targets": {
+                    "chrome": 100
```
Comment (tomasklim) [inline packages/components/.babelrc.json]: In Suite, we do not use `@babel/preset-env`, so I guess we should not use it here

Comment (tomasklim) [reply packages/components/.babelrc.json]: It should use the same config as we have in `base.webpack.config.js`

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: packages/components/.babelrc.json (hunk 3 above)
Comment (dahaca) [reply packages/components/.babelrc.json]: True, this got added automatically by `npx storybook@latest upgrade` I believe

Comment (dahaca) [reply packages/components/.babelrc.json]: bf809300b57c3b0f40b84cea34d82ea82c01c9e4

File: packages/components/package.json (hunk 2 above)
Comment (dahaca) [reply packages/components/package.json]: bf809300b57c3b0f40b84cea34d82ea82c01c9e4

**Review by tomasklim — APPROVED** [review]

**Pull request merged**